### PR TITLE
server: fix http schema table_id api with partition ID

### LINKED ISSUE #37844: Get  schema by table_id not work for partition
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

```sql
create table t1 (id int KEY)
		partition by range (id) (
		PARTITION p0 VALUES LESS THAN (3),
		PARTITION p1 VALUES LESS THAN (5),
		PARTITION p2 VALUES LESS THAN (7),
		PARTITION p3 VALUES LESS THAN (9))
```
curl http://127.0.0.1:10080/schema?table_id=x where the x is partition table ID of p0.

### 2. What did you expect to see? (Required)
Get the table info.

### 3. What did you see instead (Required)
Table which ID = %s does not exist.

### 4. What is your TiDB version? (Required)
a3bb331f3a053407384be7e3614d80ee76016dab

### PR DESCRIPTION
**Pull request by jackysp**
Signed-off-by: Jack Yu <jackysp@gmail.com>

### What problem does this PR solve?

Issue Number: close #37844

Problem Summary:

### What is changed and how it works?
Support partition table for the schema API.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by crazycs520 — APPROVED** [review]

File: server/http_handler.go [hunk 1]
```diff
@@ -1200,7 +1200,13 @@ func (h schemaHandler) ServeHTTP(w http.ResponseWriter, req *http.Request) {
...
 			return
 		}
-		writeError(w, infoschema.ErrTableNotExists.GenWithStack("Table which ID = %s does not exist.", tableID))
+		// The tid maybe a partition ID of the partition-table.
+		tbl, _, _ := schema.FindTableByPartitionID(int64(tid))
+		if tbl == nil {
+			writeError(w, infoschema.ErrTableNotExists.GenWithStack("Table which ID = %s does not exist.", tableID))
+			return
+		}
+		writeData(w, tbl.Meta())
```
Comment (mjonss) [inline server/http_handler.go:1209]: To keep the logic more similar to above.
```suggestion
		if tbl != nil {
			writeData(w, tbl.Meta())
			return
		}
		writeError(w, infoschema.ErrTableNotExists.GenWithStack("Table which ID = %s does not exist.", tableID))
```

**Review by mjonss — APPROVED** [review]
LGTM, just a minor suggestion

File: server/http_handler.go (hunk 1 above)
Comment (jackysp) [reply server/http_handler.go:1210]: Sorry, I missed this suggestion. Maybe I could refine the logic style in another PR.

**Pull request merged**